core/web/resolver/job_proposal.go

```diff
@@ -0,0 +1,257 @@
+package resolver
+
+import (
+	"context"
+	"database/sql"
+	"strconv"
+
+	"github.com/graph-gophers/graphql-go"
+	"github.com/pkg/errors"
+
+	"github.com/smartcontractkit/chainlink/core/services/feeds"
+	"github.com/smartcontractkit/chainlink/core/web/loader"
+)
+
+var notFoundErrorMessage = "job proposal not found"
+
+type JobProposalStatus string
+
+const (
+	PENDING   JobProposalStatus = "PENDING"
+	APPROVED  JobProposalStatus = "APPROVED"
+	REJECTED  JobProposalStatus = "REJECTED"
+	CANCELLED JobProposalStatus = "CANCELLED"
+)
```

Comment:
Are these any different than the constants defined in the `feeds` package? Maybe we can just use those and free ourselves of some code duplication?

Reply:
I believe we need to do this to support the enum type in `graphql` but I could be wrong.

Reply:
Yes the reason here is to support the enum type in `graphql`.